src/bin/generate_docs/generation_strategy/parsing_utils.rs

```diff
@@ -0,0 +1,195 @@
+    parser: &SchemaParser<'_, S>,
+    schema: SchemaObject,
+) -> Kind {
+    let map = schema.object.unwrap();
+    let value_type = parser.parse_schema_object(map.additional_properties.unwrap().into_object());
+
+    Kind::Map {
+        value_type: Box::new(value_type),
+    }
+}
+
+pub fn parse_object<S: GenerationStrategy>(
+    parser: &SchemaParser<'_, S>,
+    schema: SchemaObject,
+) -> Kind {
+    let object = schema.object.unwrap();
+    let mut fields = Vec::new();
+
+    for (name, prop) in object.properties {
+        let field = parser.parse_schema_object(prop.into_object());
+        fields.push((name, field));
+    }
+
+    Kind::Object { fields }
+}
+
+/// Flattens nested unions into a single union
+pub fn flatten_unions(definitions: &mut [TopLevelDefinition]) {
+    fn flatten(def: &mut TypeDefinition) {
+        let Kind::Union(variants) = &mut def.kind else {
+            return;
+        };
+
+        let mut variants_to_merge = Vec::new();
+        let mut variants_to_remove = Vec::new();
+        for (i, variant) in variants.iter_mut().enumerate() {
+            flatten(variant);
+
+            if let Kind::Union(mut subvariants) = variant.kind.clone() {
+                variants_to_merge.append(&mut subvariants);
+                variants_to_remove.push(i);
+            }
+        }
+
+        // Remove variants that were merged
+        for i in variants_to_remove.into_iter().rev() {
+            variants.remove(i);
+        }
+
+        variants.append(&mut variants_to_merge);
+    }
+
+    for TopLevelDefinition { definition, .. } in definitions.iter_mut() {
+        flatten(definition);
+    }
+}
+
+/// Inlines definitions that are defined in config as always inlined or are inlineable (simple types, such as strings, numbers, etc.)
+pub fn inline_definitions(definitions: &mut Vec<TopLevelDefinition>, config: &DocsConfig) {
+    fn inline_for(
+        definition: &mut TypeDefinition,
+        inlineable_definitions: &HashMap<String, TypeDefinition>,
+    ) {
+        match &mut definition.kind {
+            Kind::Ref(reference) => {
+                if let Some(inline_def) = inlineable_definitions.get(reference) {
+                    *definition = definition.merge_into(inline_def);
+                }
+            }
+            Kind::Tuple(variants) | Kind::Union(variants) => {
+                variants
+                    .iter_mut()
+                    .for_each(|variant| inline_for(variant, inlineable_definitions));
+            }
+            Kind::Array { array_type } => inline_for(array_type, inlineable_definitions),
+            Kind::Object { fields } => {
+                fields
+                    .iter_mut()
+                    .for_each(|(_, prop)| inline_for(prop, inlineable_definitions));
+            }
+            Kind::Map { value_type } => inline_for(value_type, inlineable_definitions),
+            _ => {}
+        }
+    }
+
+    let mut inlineable_definitions = HashMap::new();
```

Comment:
inlinable means that sth can be inlined but that is not the way you are using this value

inlined_definitions or definitions_to_inline